gensound/transforms.py

```diff
@@ -187,34 +187,49 @@ def realise(self, audio):
 
 ######### Level/ampltidue Stuff ###################
 
-class Fade(Transform):
-    min_fade = -50
-    
-    def __init__(self, is_in=True, duration=3e3):
-        self.is_in = is_in
+class Fade(Transform):   
+    def __init__(self, duration=None, curve=None, is_in=True, degree=1):
+        assert curve in ("linear", "polynomial")
+        # default setting for backwards compatibility (allowing usage of both Fade() and FadeIn/FadeOut()
+        if curve == None: curve = "linear"
+        if duration == None: duration = 3e3
```

Comment:
Python already has a syntax for specifying default values for arguments, it's best to use it instead of handling it manually

Reply:
Hi, have you seen my comment above? 

I thought about it and preferred this way! I'll explain, and if you aren't convinced - I'll change. 
Although `Fade` is not the preferred way, I still want the have it ready to be called by the user.
So, the defaults values have to be in three different places: `Fade()`, `FadeIn()`,`FadeOut()`, and it doesn't look as a good practice to me to have the default set in three separate places. Any decision of change later will have to be manually managed in 3 different spots.  
So I have set all the defaults to `None` in all those places. That is 1.  easier to maintain - no need to decided the default of each new paramer of each spot in the future - set all defaults to 'None' and have one place to take care on them all.
 2. Having any default at all allows the user to not specify any parameter if he doesn't wish to.

So this way, Most parameters set to `None`,  and initialize in the same place.

However, I agree that the `degree = 1` can be set as default in the arguments (since it is very simple and  unlikely to default to different value)
Although I have a very small experience, this initialization 
```
        if curve == None: curve = "linear"
        if duration == None: duration = 3e3
```
does look as a better practice in my opinion, so it is in one place only, in case a change of default is wanted down the road.

What do you think? :)